apis/v1alpha1/types.go

```diff
@@ -37,24 +37,53 @@ var (
 //   - For information about creating failover resource record sets in a private
 //     hosted zone, see Configuring Failover in a Private Hosted Zone (https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/dns-failover-private-hosted-zones.html).
 type AliasTarget struct {
-	DNSName      *string `json:"dnsName,omitempty"`
-	HostedZoneID *string `json:"hostedZoneID,omitempty"`
+	DNSName              *string `json:"dnsName,omitempty"`
+	EvaluateTargetHealth *bool   `json:"evaluateTargetHealth,omitempty"`
+	HostedZoneID         *string `json:"hostedZoneID,omitempty"`
+}
+
+// A complex type that identifies a CIDR collection.
+type CIDRCollection struct {
+	ID *string `json:"id,omitempty"`
+}
+
+// The object that is specified in resource record set object when you are linking
+// a resource record set to a CIDR location.
+//
+// A LocationName with an asterisk “*” can be used to create a default CIDR
+// record. CollectionId is still required for default record.
+type CIDRRoutingConfig struct {
+	CollectionID *string `json:"collectionID,omitempty"`
+	LocationName *string `json:"locationName,omitempty"`
+}
+
+// The information for each resource record set that you want to change.
+type Change struct {
+	Action *string `json:"action,omitempty"`
+	// Information about the resource record set to create or delete.
+	ResourceRecordSet *ResourceRecordSet `json:"resourceRecordSet,omitempty"`
 }
 
 // The information for a change request.
 type ChangeBatch struct {
-	Comment *string `json:"comment,omitempty"`
+	Changes []*Change `json:"changes,omitempty"`
+	Comment *string   `json:"comment,omitempty"`
 }
```

Comment:
If i understand well this field is not really be provided by the user. We can deduce the necessary changes we need to make from CR deltas, and generate the right `Changes` object.